TempoIQ/Models/UpsertResponse.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using System.Linq;
+using TempoIQ.Json;
+
+namespace TempoIQ.Models
+{
+    public enum DeviceState {
+        Existing,
+        Modified,
+        Created
+    }
+
+    public class DeviceStatus
+    {
+        [JsonConverter(typeof(DeviceStateConverter))]
+        public DeviceState State { get; set; }
+        public bool Success { get; set; }
+        public String Message { get; set; }
+    }
+
+    public class UpsertResponse : Dictionary<String, DeviceStatus>
+    {
+        [JsonConstructor]
+        public UpsertResponse(IDictionary<String, DeviceStatus> data) : base()
+        {
+            foreach(var pair in data)
+                this.Add(pair.Key, pair.Value);
+        }
+         
+        /*public UpsertResponse(Dictionary<String, DeviceStatus> statuses
+        {
+            if (statuses == null)
+                throw new BadImageFormatException("YOU SUCK");
+            this.statuses = statuses;
+        }*/
+
+        public bool Success
+        {
+            get
+            {
+                foreach (DeviceStatus status in this.Values) 
```

Comment:
If you ever have to do something like this in the future, you might like to know about: https://msdn.microsoft.com/en-us/library/vstudio/bb548541%28v=vs.100%29.aspx